### DISCUSSION TITLE
Error resolving stylesheet with sass-loader with import/require keys in package.json exports

### DISCUSSION THREAD

**Discussion by karlshea** [webpack/webpack#18866] [Q&A] [answered]
I'm importing stylesheets in my main SCSS file from several different packages (like `@import "react-day-picker/style.css";`), and all of them are working except for react-day-picker. The only difference is that react-day-picker's exports are slightly different, it looks like this:

```
"exports": {
  "./style.css": {
    "import": {
      "types": "./style.css.d.ts",
      "default": "./style.css"
    },
    "require": {
      "types": "./style.css.d.ts",
      "default": "./style.css"
    }
  },
```

I am **not** passing `false` to the sass-loader `webpackImporter` option, so it should be using webpack's module resolution, and the docs for that [here](https://webpack.js.org/configuration/resolve/) would seem that should be fine? However the build fails saying the file isn't exported from the package.

There's a PR https://github.com/gpbl/react-day-picker/pull/2547 to remove the `import`/`require` keys from react-day-picker's exports for its stylesheets, but the test runner is failing when generating the documentation website. It looks like those keys are required when trying to import the stylesheet directly from a JS component.

What might be going on here?

**Comment by alexander-akait** [comment, marked as answer]
`"./style.css"` is not JS modules and can't have `"types": "./style.css.d.ts",` (only `JS` files can have,  except you have a bundler which allow `import "style.css"`, webpack allows you to do this, just a note)

When `sass-loader` try to `import` sass files it uses `sass`/`style`:
```
"exports": {
  "./style.css": {
    "sass": {
      "default": "./style.css"
    },
    "import": {
      "types": "./style.css.d.ts",
      "default": "./style.css"
    },
    "require": {
      "types": "./style.css.d.ts",
      "default": "./style.css"
    }
  },
}
```

or 

```
"exports": {
  "./style.css": {
    "style": {
      "default": "./style.css"
    },
    "import": {
      "types": "./style.css.d.ts",
      "default": "./style.css"
    },
    "require": {
      "types": "./style.css.d.ts",
      "default": "./style.css"
    }
  },
}
```

`sass` is not using `import`/`require` because it is not JS

**Reply by karlshea** [reply]
Ahhh that makes sense, thanks! `style` worked for me, but `sass` didn't.

**Reply by alexander-akait** [reply]
It means you have `.css` files and sass convert it to `@import "style.css"` and then `css-loader` resolve it
